purchase_order_supplier_return/tests/test_purchase_order_supplier_return.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.tests.common import TransactionCase
+
+
+class TestPurchaseOrderSupplierReturn(TransactionCase):
+    def setUp(self):
+        super().setUp()
+
+        self.product = self.env["product.product"].create(
+            {"name": "Product Test", "list_price": 5.0}
+        )
+        self.partner = self.env["res.partner"].create(
+            {
+                "name": "Partner",
+            }
+        )
+        self.purchase_order = self.env["purchase.order"].create(
+            {
+                "partner_id": self.partner.id,
+            }
+        )
+        self.purchase_order_line_1 = self.env["purchase.order.line"].create(
+            {
+                "order_id": self.purchase_order.id,
+                "product_id": self.product.id,
+                "product_qty": -1,
+            }
+        )
+        self.purchase_order_line_2 = self.env["purchase.order.line"].create(
+            {
+                "order_id": self.purchase_order.id,
+                "product_id": self.product.id,
+                "product_qty": 1,
+            }
+        )
+
+    def test_negative_purchase_order_line(self):
+        """Test that the negative purchase_order_line is correctly received
+        with the negative quantity when the related picking is validated."""
+        self.purchase_order.button_confirm()
+        assert self.purchase_order_line_1.qty_received == 0
+        assert self.purchase_order_line_1.qty_received == 0
+
+        # Validate the pickings related to the purchase order
+        for picking in self.purchase_order.picking_ids:
+            picking.action_set_quantities_to_reservation()
+            picking.button_validate()
+
+        self.purchase_order.order_line._compute_qty_received()
```

Comment:
if you need to call compute here there might be an issue... what is the reason?

Reply:
To be honest I don't know why method ` _compute_qty_received` is not called on the unit test.
I refactored the code to update the quantity received in `move._action_done`.